### PR TITLE
Delete obs-ndi plugin on macos after installing DistroAV

### PR DESCRIPTION
**Pull request by BitRate27**
Users were instructed to manually delete the obs-ndi.plugin from obs-studio before installing DistroAV. This proved to be a challenge for some users and is prone to user error. With this PR, the deletion of the old plugin is done automatically as part of the installation of the DistroAV plugin.

Target Usage:

- MacOS package installer

The PR solves this issue by:

- Modifying the existing UninstallOBS-NDI.sh script to also delete the plugin for all users
- Add the script to the MacOS package
- Execute the script after installation of DistroAV, causing the deletion of the obs-ndi.plugin

Testing:

- Was only able to test the script deletes the plugin.
- **Since I cannot build on MacOS, delivery and execution of the script when the package is installed was NOT tested.**

### CODE REVIEW

**Comment by Trouffman** [comment]
I will be testing this for Mac OS.

**Comment by Trouffman** [comment]
This will NOT remove files in the system library as this is "offlimit" for scripts.
Files should not be in there anyway.

File: build-aux/postinstall
Comment (paulpv) [inline build-aux/postinstall:1]: I wouldn't put this in the build-aux folder. 
Seems like directly in `cmake/macos/resources` would be a better place.

**Comment by paulpv** [comment]
> This will NOT remove files in the system library as this is "offlimit" for scripts. Files should not be in there anyway.

So why does it even try?

**Comment by paulpv** [comment]
Who has a problem with this? 
1% of our users? 
Can't they just learn how to read?

**Comment by Trouffman** [comment]
> > This will NOT remove files in the system library as this is "offlimit" for scripts. Files should not be in there anyway.
> 
> So why does it even try?

Couple things there :
- The script will not be able to access the systems Application Support folder by default. (No access unless ran as admin or root or else

- But, if users ended up with files there this means the script should also have access. )I could not fully test that..

You review is important, we could only target the OBS-ndi files in the user ~/Library.


Note that there is another PR adding the path of the detected OBS-ndi that should help most requests.


This was the main source for support requests coming in the last weeks as in "I followed the install instructions but it still says it's there".

**Comment by paulpv** [comment]
Understood. Just seems like a lot of work for "a few" users.

Still, it is nice to have I guess. I just wouldn't want it to cause more problems than it is trying to fix.

I wouldn't want it to fix a problem for 3 users [that can't/won't read/follow directions] but cause a problem for hundreds.

**Comment by paulpv** [comment]
I would still like to move the `postinstall` script from `build-aux` to `cmake/macos/resources`

File: build-aux/postinstall
Comment (Trouffman) [reply build-aux/postinstall:1]: fixed as suggested.

**Review by paulpv — APPROVED** [review]

File: cmake/windows/resources/installer-Windows.iss.in [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#define MyAppName "@CMAKE_PROJECT_NAME@"
+#define MyAppName "@PLUGIN_DISPLAY_NAME@"
```
Comment (paulpv) [inline cmake/windows/resources/installer-Windows.iss.in:1]: Don't do this in this PR that never touches this file.
Do this in https://github.com/DistroAV/DistroAV/pull/1134 that already touches this file.

Comment (Trouffman) [reply cmake/windows/resources/installer-Windows.iss.in]: my bad had the wrong branch open -_-. Fixed.

**Review by Trouffman — APPROVED** [review]

**Comment by Trouffman** [comment]
@BitRate27 I see that the postinstall & script folder is not being deleted after distroav install.

I am not keen on creating garbage files, we should revisit this approach.

**Comment by BitRate27** [comment]
Thanks! Will not be able to look at this until Dec 2.

**Pull request merged**